front/production_checks/checks/scrub_deleted_core_document_versions.ts

```diff
@@ -146,32 +159,45 @@ async function scrubDocument(
 
   const path = `${dataSource.project}/${dataSource.internal_id}/${documentIdHash}/${hash}`;
 
-  const storage = new Storage({ keyFilename: SERVICE_ACCOUNT });
-
   const [files] = await storage
     .bucket(DUST_DATA_SOURCES_BUCKET)
     .getFiles({ prefix: path });
 
-  await Promise.all(
-    files.map((f) => {
-      if (!seen.has(f.name)) {
-        seen.add(f.name);
-        logger.info(
-          {
-            path: f.name,
-            documentId: document_id,
-            documentHash: hash,
-            dataSourceProject: dataSource.project,
-            dataSourceInternalId: dataSource.internal_id,
-            dataSourceId: dataSource.id,
-          },
-          "Scrubbing"
-        );
-
-        return f.delete();
-      }
-    })
-  );
+  try {
+    await Promise.all(
+      files.map((f) => {
+        if (!seen.has(f.name)) {
+          seen.add(f.name);
+          logger.info(
+            {
+              path: f.name,
+              documentId: document_id,
+              documentHash: hash,
+              dataSourceProject: dataSource.project,
+              dataSourceInternalId: dataSource.internal_id,
+              dataSourceId: dataSource.id,
+            },
+            "Scrubbing"
+          );
+
+          return f.delete();
+        }
+      })
+    );
+  } catch (e) {
```

Comment:
Why don't we jus de-dup the deletion to make sure we only send one delete call?
Having those "expected errors" there put as at risk of hiding some unexecpected ones.

Reply:
Removed. Works as expected assuming only one check is run at a time